pkgs/by-name/fl/flashprog/package.nix

```diff
@@ -14,12 +14,12 @@
 
 stdenv.mkDerivation {
   pname = "flashprog";
-  version = "1.0.1";
+  version = "1.1";
 
   src = fetchgit {
     url = "https://review.sourcearcade.org/flashprog";
-    rev = "2ca11f9a4101ea230081d448ab2b570425b7f0bd";
-    hash = "sha256-pm9g9iOJAKnzzY9couzt8RmqZFbIpKcO++zsUJ9o49U=";
+    rev = "20dc8e2c0e04b06db2cc6a3b9072917bff11bee5";
```

Comment:
This is not the [p1.1 tag](https://review.sourcearcade.org/plugins/gitiles/flashprog/+/refs/tags/p1.1), but an untagged later commit, that seems to be the starting point for the 1.1.x branch (without any functional difference though, AFAICS). What's the reason for that?

Reply:
See my other comment. Also from the PR description:

> A WIP patch in preparation to the next flashprog release.

Nice to see you here by the way :)

Reply:
The referenced commit includes the version change to 1.1 in source code while the p1.1 tag is the point of branching of. See https://review.sourcearcade.org/plugins/gitiles/flashprog/+log/refs/heads/1.1.x/ for reference.

Reply:
> The referenced commit includes the version change to 1.1 in source code while the p1.1 tag is the point of branching of. See https://review.sourcearcade.org/plugins/gitiles/flashprog/+log/refs/heads/1.1.x/ for reference.


Ok, the structure I see in git is like this:

```   
      p1.1
------- o ----- o ------ o ---...    main
          \
           o --------- ...           1.1.x
```

So I would expect:

 - p1.1 is the 1.1 release (because it's tagged)
 - untagged commits are not releases, and won't be used as such
 - further releases such as p1.1.1 will be made on the 1.1.x branch
 - perhaps a p1.1.0 commit will be made on the 1.1.x branch? I don't know


So my irritation/confusion is just that an untagged commit is used as a release, when a tagged commit is right next to it. I'm probably missing something though, hence my questions.

Reply:
> Not sure which conclusion to draw from that, does it mean you want to change the `rev` later before marking the PR as non-draft?

I also think it would be good to change the rev of this PR to the v1.1 tag as soon it is available.

Reply:
> I also think it would be good to change the rev of this PR to the v1.1 tag as soon it is available.

@felixsinger I would humbly suggest using a TODO item (aka. checkbox) in the PR description for that. They make unfinished tasks very easy to spot.

Sorry for the noise, everyone :)

Reply:
> When released there will be an tag v1.1 which will indicate the released commit. The tag p1.1 indicates where the branch of happened. See the history for v1.0 as reference: https://review.sourcearcade.org/plugins/gitiles/flashprog/+log/2ca11f9a4101ea230081d448ab2b570425b7f0bd

I've tried to document the branching and release policy upstream: https://review.sourcearcade.org/c/flashprog/+/127/1/README

Reply:
I usually make a WIP (like this one) patch as soon as I know there is an update coming. So I pick the most recent commit from time to time and push it, so that people and myself can test.

Version 1.1 just got officially released and I've updated this patch to the corresponding commit.